src/components/Avatar.js

```diff
@@ -60,71 +60,69 @@ const defaultProps = {
     name: '',
 };
 
-class Avatar extends PureComponent {
-    constructor(props) {
-        super(props);
-        this.state = {
-            imageError: false,
-        };
-    }
+function Avatar(props) {
+    const [imageError, setImageError] = useState(false);
+    const prevNetworkStatusRef = useRef(props.network.isOffline);
 
-    componentDidUpdate(prevProps) {
-        const isReconnecting = prevProps.network.isOffline && !this.props.network.isOffline;
-        if (!this.state.imageError || !isReconnecting) {
+    useEffect(() => {
+        const isReconnecting = prevNetworkStatusRef.current && !props.network.isOffline;
+        if (!imageError || !isReconnecting) {
             return;
         }
-        this.setState({imageError: false});
-    }
+        setImageError(false);
+    }, [props.network.isOffline]);
```

Comment:
Started a conversation around whether or not `imageError` should be include in the dep array or not in the open-source channel here: https://expensify.slack.com/archives/C01GTK53T8Q/p1681117333835879

Reply:
Added a comment to explain why `imageError` shouldn't be a dependency in running side effects.